brave/brave-ios
# Ref #2383: Adds SKU Purchase UI

## Summary of Changes

This pull request refs #2383 

## Submitter Checklist:


## Test Plan:

- Prefix: If building locally, ensure you remove `NO_SKUS` compiler flag from `Base.xcconfig`
- With rewards disabled visit store, attempt to buy with BAT, verify brings up enable rewards popup
- After rewards is enabled, open buy with BAT again, verify not enough funds appears
- Claim grant, open buy with BAT again, verify buy button appears
- Click buy, verify loading, and completion screen appears
- Open panel, verify BAT has been removed from your wallet (currently just tips publisher)

## Screenshots:

<details>
<summary>Rewards Disabled</summary>

![Simulator Screen Shot - iPhone 8 - 2020-03-19 at 13 58 43](https://user-images.githubusercontent.com/529104/77115200-464baf00-6a04-11ea-9993-6427648294a7.png)

</details>

<details>
<summary>Not enough BAT</summary>

![Simulator Screen Shot - iPhone 11 Pro - 2020-03-19 at 17 06 39](https://user-images.githubusercontent.com/529104/77115328-81e67900-6a04-11ea-93f4-29a33312c533.png)

</details>


<details>
<summary>Standard</summary>

![Simulator Screen Shot - iPhone 11 Pro - 2020-03-19 at 13 56 14](https://user-images.githubusercontent.com/529104/77115410-a6425580-6a04-11ea-9cb9-c52b4214b0f7.png)

</details>

<details>
<summary>Loading, Completion</summary>

![Simulator Screen Shot - iPhone 11 Pro - 2020-03-19 at 13 56 17](https://user-images.githubusercontent.com/529104/77115372-9460b280-6a04-11ea-9713-eba95968e565.png)
![Simulator Screen Shot - iPhone 11 Pro - 2020-03-19 at 13 56 19](https://user-images.githubusercontent.com/529104/77115375-96c30c80-6a04-11ea-895b-b8f81772db69.png)

</details>

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
